sopel/bot.py

```diff
@@ -530,6 +530,9 @@ def dispatch(self, pretrigger):
                                 match = True
                         if not match:
                             continue
+                    if (not (hasattr(func, 'echo') and func.echo is True) and
+                            trigger.nick.lower() == self.nick.lower()):
+                        continue
```

Comment:
I'd put this check first in the enclosing `for` loop, but the order isn't really an issue tbh.